### PR TITLE
Checking nil contexts in fctx

### PR DESCRIPTION
**Pull request by semihbkgr**
Add `nil` context checks to `fctx` package.

### CODE REVIEW

File: fault.go [hunk 1]
```diff
@@ -21,10 +21,10 @@ func Wrap(err error, w ...Wrapper) error {
 		return nil
 	}
 
-	// the passed err might already have a location if it's a fault.New error, or it might not if it's another type of
+	// The passed err might already have a location if it's a fault. New error, or it might not if it's another type of
```
Comment (Southclaws) [inline fault.go:24]: this isn't a sentence boundary period, it's a Go source code snippet, as in `fault.New("some error")`

Comment (Southclaws) [reply fault.go:24]: quotes would make it more clear tbh

**Comment by Southclaws** [comment]
Thanks for this fix! Just a small comment regarding the change of the `fault.New` in the comment and I'm happy to merge!

File: fault.go (hunk 1 above)
Comment (semihbkgr) [reply fault.go:24]: Oh, I see

**Comment by semihbkgr** [comment]
should be ready to merge!

**Review by Southclaws — APPROVED** [review]
Thanks!

**Pull request merged**